Sources/SpeziScheduler/Notifications/SchedulerNotificationsConstraint.swift

```diff
@@ -16,16 +16,16 @@ import UserNotifications
 /// ```swift
 /// actor MyActor: Standard, SchedulerNotificationsConstraint {
 ///     @MainActor
-///     func notificationContent(for task: borrowing Task, content: borrowing UNMutableNotificationContent) {
+///     func updateNotificationContent(for event: borrowing Event, content: borrowing UNMutableNotificationContent) {
 ///         content.subtitle = "Complete the Questionnaire"
 ///     }
 /// }
 /// ```
 public protocol SchedulerNotificationsConstraint: Standard {
-    /// Customize the  notification content of a notification for the event of a task.
+    /// Customize the notification content of a notification for the event of a task.
     /// - Parameters:
-    ///   - task: The task for which we generate the notification for.
+    ///   - task: The ``Event`` for which we generate the notification for.
     ///   - content: The default notification content generated by ``SchedulerNotifications`` that can be customized.
     @MainActor
-    func notificationContent(for task: borrowing Task, content: borrowing UNMutableNotificationContent)
+    func updateNotificationContent(for event: borrowing Event, content: borrowing UNMutableNotificationContent)
```

Comment:
As far as I can see this would be a breaking change. Can we mitigate this in some way or would it make sense to tag a version 3.0 (maybe a beta first) for this?

Reply:
i was thinking about this as well. if it weren't a protocol we could simply add a deprecated overload and forward the call but the fact that it's a protocol makes this a bit more difficult.

Reply:
update i played around with this a little bit and i think technically we might be able to make it work but it wouldn't really be pretty. IMO it might just be easier to turn this into a breaking change

Reply:
Hmm, it would be ideal if we can make it a non-breaking version bump to make it as easy as possible to update and resolve related packages. A default implementation in the protocol might do its job here? Or we go ahead and create a new protocol and dispatch on each of them depending on the protocol implemented by the scheduler. If it his too painful we can make it a breaking change, but I would love to explore if we can avoid it and/or package it with more breaking changes down the line.

Reply:
@PSchmiedmayer i ended up rolling this back entirely.

changing it from a `Task` to an `Event` simply doesn't work well since some notifications are scheduled at task-level and some at event-level. we could of course have two functions but that'd be overkill.

i still wanna change the name at some point, but we should simply do that at the next breaking change release. we're tracking this in #59